Automattic/wp-calypso
# Fix error handling in the install-if-deps-outdated script

Fixes invalid assignment of `spawnSync().status` to `installResult`. We want to assign and check the whole result object, not just its `status` field.

If `npm ci` failed, the error check code failed to detect it and proceeded to the `fs.utimesSync` call, even if `node_modules` didn't exist.

Also, remove the `-s` argument from `npm run` when `install-if-deps-outdated`. The `-s` argument only hides errors, while stdout continues to be full of output from `npx` and `lerna`.

**How to test:**
Remove `node_modules` and `package-lock.json`. For example, a failed run of `npm run update-deps` will leave your project in such a state.

Run `npm start`. Verify that `npm start` runs the `install-if-deps-outdated` script and reports errors correctly -- i.e., `npm ci` complaining about missing `package-lock.json`.

Before this patch, the error you got was about `utime` not finding a `node_modules` folder 🤷‍♂
